File: src/main/java/net/dashflight/data/postgres/DashflightPostgresConnectionOptionProvider.java
package net.dashflight.data.postgres;

import net.dashflight.data.config.ConfigValue;
import net.dashflight.data.config.Configurable;

/**
 * Pulls the required postgres connection options from s3
 */
class DashflightPostgresConnectionOptionProvider implements PostgresConnectionOptionProvider, Configurable {

    private static final String APP_NAME = "java-postgres";


    @ConfigValue("pg_host")
    private String host;

    @ConfigValue("pg_port")
    private int port;

    @ConfigValue("pg_dbname")
    private String dbname;

    @ConfigValue("pg_username")
    private String username;

    @ConfigValue("pg_password")
    private String password;

    @ConfigValue("application_name")
    private String applicationName = "Dashflight Java App";

    @ConfigValue("max_pool_size")
    private int maxPoolSize;


    public DashflightPostgresConnectionOptionProvider() {
        registerWith(APP_NAME);
    }


    @Override
    public PostgresConnectionOptions get() {
        return PostgresConnectionOptions.builder()
                .host(host)
                .port(port)
                .dbname(dbname)
                .username(username)
                .password(password)
                .applicationName(applicationName)
                .maxPoolSize(maxPoolSize)
                .build();
    }
}

File: src/main/java/net/dashflight/data/postgres/DashflightPostgresConnectionOptionProvider.kt
package net.dashflight.data.postgres

import hydro.engine.Hydro.hydrate

/**
 * Pulls the required postgres connection options from s3
 */
class DashflightPostgresConnectionOptionProvider(applicationName: String) : PostgresConnectionOptionProvider {

    private val host: String? by hydrate("pg_host")

    private val port: Int by hydrate("pg_port")

    private val dbname: String? by hydrate("pg_dbname")

    private val username: String? by hydrate("pg_username")

    private val password: String? by hydrate("pg_password")

    private val _applicationName: String? by hydrate("application_name", applicationName)

    private val maxPoolSize: Int by hydrate("max_pool_size")

    override fun get(): PostgresConnectionOptions {
        return PostgresConnectionOptions(host!!, port, dbname, username, password, _applicationName ?: "Dashflight Java App", maxPoolSize)
    }
}